include/crocoddyl/core/integrator/euler.hxx

```diff
@@ -96,24 +99,26 @@ void IntegratedActionModelEulerTpl<Scalar>::calcDiff(const boost::shared_ptr<Act
 
   // Computing the derivatives for the time-continuous model (i.e. differential model)
   differential_->calcDiff(d->differential, x, u);
-  differential_->get_state()->Jintegrate(x, d->dx, d->dxnext_dx, d->dxnext_ddx);
 
-  d->Fx = d->dxnext_dx;
   if (enable_integration_) {
     const MatrixXs& da_dx = d->differential->Fx;
     const MatrixXs& da_du = d->differential->Fu;
-    d->ddx_dx << da_dx * time_step_, da_dx;
-    d->ddx_du << da_du * time_step_, da_du;
-    for (std::size_t i = 0; i < nv; ++i) {
-      d->ddx_dx(i, i + nv) += 1.;
-    }
-    d->Fx.noalias() += time_step_ * (d->dxnext_ddx * d->ddx_dx);
-    d->Fu.noalias() = time_step_ * (d->dxnext_ddx * d->ddx_du);
-    d->Lx = time_step_ * d->differential->Lx;
-    d->Lu = time_step_ * d->differential->Lu;
-    d->Lxx = time_step_ * d->differential->Lxx;
-    d->Lxu = time_step_ * d->differential->Lxu;
-    d->Luu = time_step_ * d->differential->Luu;
+    d->Fx.topRows(nv).noalias() = da_dx * time_step2_;
+    d->Fx.bottomRows(nv).noalias() = da_dx * time_step_;
+    d->Fx.topRightCorner(nv, nv).diagonal().array() += Scalar(time_step_);
+
+    d->Fu.topRows(nv).noalias() = da_du * time_step2_;
+    d->Fu.bottomRows(nv).noalias() = da_du * time_step_;
+
+    differential_->get_state()->JintegrateTransport(x, d->dx, d->Fx, second);
```

Comment:
I have generated the log files for all the examples, and I encountered that this branch produces much more iterations.

I believe the source of problem could be here, since I am not sure about this implementation.

Unfortunately, we don't have the unittest code for the euler integrator yet.

I asked you two things:
 1. could you validate that the output of this routines has the same numerical value that before?
 2. could you implement unittest code for the `JintegrateTransport` (see `test_state.cpp`)?

Reply:
Define much more iterations please?

Reply:
On which examples / what is the before and after?

Reply:
> Define much more iterations please?

From tens to hundreds. There is a bug in this PR, and I suspected the problem is here, but it is hard to tell you with so many changes.

Please do not add more stuff to this PR, it is taken so much time and it needs a lot of more validation from my side. Unfortunately, there are pieces of the library that are not tested yet.

Reply:
Once I fixed the Python bindings, you could try yourself @proyan.

Reply:
Which examples? and what is the iterations?

Reply:
All problems that involve contact dynamics. I will share with you (through riot) the log files that I generated, the worse case is for the `bipedal_walk*` files.

Reply:
The problem was not in this line. It was a missing calculation for terminal model. It is fixed now. I'll close this.